source/distributed/shared_tria.cc

```diff
@@ -134,6 +135,20 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      Assert ((dynamic_cast<const dealii::parallel::distributed::Triangulation<dim,spacedim> *>(&old_tria) == NULL),
+              ExcMessage("Cannot use this function on parallel::distributed::Triangulation."));
```

Comment:
It still would be better to assert you get what you want (base class or shared) because we might introduce another Triangulation class at some point.